repository/src/main/java/com/epam/esm/exceptions/BaseCustomException.java

```diff
@@ -0,0 +1,17 @@
+package com.epam.esm.exceptions;
+
+import com.fasterxml.jackson.annotation.*;
+import lombok.Getter;
+
+import java.io.IOException;
+
+@Getter
+@JsonIgnoreProperties({"stackTrace", "cause", "suppressed", "localizedMessage"})
+public abstract class BaseCustomException extends IOException {
```

Comment:
Why here did you use IOException? Is the IOExeption base exception for all exceptions? I would advise to use RuntimeException.